### DISCUSSION TITLE
How do I update wingetUI?

### DISCUSSION THREAD

**Discussion by aaronliu0130** [Devolutions/UniGetUI#452] [Q&A] [answered]
I can't find any buttons to update it and it says it's updating in the background when I X'd it but it never restarted

**Comment by marticliment** [comment, marked as answer]
updating was broken not that much ago, and while winget was the second upgrade mechanism, tí got also broken.

To upgrade you can manually install the latest version of WingetUI from the discover software page of WingetUI itself.

**Reply by aaronliu0130** [reply]
Is there an ETA for fixing updating?

**Reply by marticliment** [reply]
No, since i don't understand why it does not work. The installer does not launch.

However, the second update mechanism has been fixed, and from version 1.5.2 and higher you can update from wingetui

**Reply by aaronliu0130** [reply]
Does the second mechanism refer to winget or the notification that it's updating?

**Reply by marticliment** [reply]
It refers to winget. WingetUI will appear as an update on wingetui, and when double-clicking it wingetui will quit and start updating in the background

**Comment by elliot-100** [comment]
Hi, I came to WingetUI while updating was broken, found this discussion and resolved the problem. 

But I noticed that there was nothing on how to update WingetUI in the README or Wiki, which was a bit of a gap.

So I've just added pull request #592 which simply adds: "_Updating ... WingetUI can be updated like any other package within WingetUI_", (i.e. normal behaviour) and fixes a problem with inconsistent heading levels.

**Reply by marticliment** [reply]
yes, wingetui does get updated like this, but no update has been released yet with the patch

**Reply by ppvnf** [reply]
If the regular updater isn't working, shouldn't it be disabled while a fix hasn't been applied or/and display a message explaining how to update WingetUI using the update section? TBH I don't think it would be a big problem if the updater only downloaded the installer, letting the user handle the install process

**Reply by marticliment** [reply]
I mean, it is not always broken, it works or not depending on a random pattern, so it helps people. it is true that it should be disabled and replaced by the winget fallback, because it works way better.

**Reply by ppvnf** [reply]
If you replace it by the Winget fallback what would happen to people who installed WingetUI through Scoop? I don't think Scoop is able to update apps that are running. Also, shouldn't this be tracked on an issue?

**Reply by aaronliu0130** [reply]
There are no Scoop packages to install WingetUI.

**Reply by ppvnf** [reply]
There's this one in the extras bucket: https://github.com/ScoopInstaller/Extras/blob/master/bucket/wingetui.json

**Reply by aaronliu0130** [reply]
My bad, it appears my internet got filtered while at work.

**Reply by marticliment** [reply]
> If you replace it by the Winget fallback what would happen to people who installed WingetUI through Scoop? I don't think Scoop is able to update apps that are running. Also, shouldn't this be tracked on an issue?

well, the scoop. bucket downloads and runs the _official_ installer, so since the installer kills wingetui when updating, it should be fine.

I have not tested it, though

**Reply by ppvnf** [reply]
It doesn't work. I've justed tested updating from WingetUI 1.5.1 to 1.5.3 using a custom bucket:

wingetui: 1.5.1 -> 1.5.3 Updating one outdated app: Updating 'wingetui' (1.5.1 -> 1.5.3) Downloading new version Downloading https://github.com/martinet101/WingetUI/releases/download/1.5.3/WingetUI.Installer.exe (44.5 MB)... Checking hash of WingetUI.Installer.exe ... ok. ERROR The following instances of "wingetui" are still running. Close them and try again. Handles  NPM(K)    PM(K)      WS(K)     CPU(s)     Id  SI ProcessName -------  ------    -----      -----     ------     --  -- ----------- 421      36    90792     132700       6.52   1964   1 WingetUI

**Reply by ppvnf** [reply]
I think the root of this problem is the fact inno isn't killing the process properly. I can't uninstall wingetui if the process is running as well

**Reply by marticliment** [reply]
Could it be that this check is done by scoop and not by the inno installer? Because WingetUI through winget updates itself from 1.5.1 to 1.5.3 flawlessly

**Reply by ppvnf** [reply]
It's probably not Inno's fault. I am not sure if Scoop runs the installer on the background like Winget does or if it simply extracts the files inside the .EXE and try to replace them with newer ones. And it might detect that the folder is being used by a program and stop the install process

**Reply by aaronliu0130** [reply]
Scoop by default installs a helper called `innounp` to unpack innosetups. https://github.com/ScoopInstaller/Scoop/search?q=innosetup So it probably extracts

**Reply by marticliment** [reply]
Ok then, that's the issue. 
They should (scoop) sort the update thing, letting running apps to be updated
